src/VisionModel/resnet.jl

```diff
@@ -0,0 +1,98 @@
+using Flux
+
+export ResNet18, ResNet34, ResNet50, ResNet101, ResNet152
+
+basicblock(inplanes, outplanes, downsample = false) = downsample ? 
+  Chain(Conv((3, 3), inplanes => outplanes[1], stride = 2),
+        BatchNorm(outplanes[1], relu),
+        Conv((3, 3), outplanes[1] => outplanes[2], stride = 1, pad = 1),
+        BatchNorm(outplanes[2], relu)) : 
+  Chain(Conv((3, 3), inplanes => outplanes[1], stride = 1),
+        BatchNorm(outplanes[1], relu),
+        Conv((3, 3), outplanes[1] => outplanes[2], stride = 1, pad = 1),
+        BatchNorm(outplanes[2], relu))
+
+bottleneck(inplanes, outplanes, downsample = false) = downsample ?
+  Chain(Conv((1, 1), inplanes => outplanes[1], stride = 2),
+        BatchNorm(outplanes[1], relu),
+        Conv((3, 3), outplanes[1] => outplanes[2], stride = 1, pad = 1),
+        BatchNorm(outplanes[2], relu),
+        Conv((1, 1), outplanes[2] => outplanes[3], stride = 1),
+        BatchNorm(outplanes[3], relu)) :
+  Chain(Conv((1, 1), inplanes => outplanes[1], stride = 1),
+        BatchNorm(outplanes[1], relu),
+        Conv((3, 3), outplanes[1] => outplanes[2], stride = 1, pad = 1),
+        BatchNorm(outplanes[2], relu),
+        Conv((1, 1), outplanes[2] => outplanes[3], stride = 1),
+        BatchNorm(outplanes[3], relu))
+
+function projection(inplanes, outplanes, stride)
+  shortcut = Chain(Conv((1, 1), inplanes => outplanes, stride=stride),
+                   BatchNorm((outplanes), relu))
+  return (x, y) -> x + shortcut(y)
+end
+
+# array -> PaddedView(0, array, outplanes) for zero padding arrays
+function identity(inplanes, outplanes, stride)
+  shortcut = array ->  convert(Array{Float32}, cat(array, zeros(eltype(array), outplanes[end] - inplanes)); dims = (1, 2, 3))
```

Comment:
I found the error. My mistake, I thought you just copied the code [from the gist I linked](https://gist.github.com/darsnack/b5ba3810fd248e1752e32986bd177962), but this cat operation is not the same. Here is the fix:
```julia
shortcut = array -> cat(array, zeros(eltype(array), size(array, 1), size(array, 2), outplanes[end] - inplanes, size(array, 4)); dims = 3)
```

Reply:
no, it didn't resolve errors.

Reply:
I think its pretty similar to my implementation.

Reply:
Yeah but your implementation creates a vector not a tensor, and it pads to the first three dimensions.